src/server/editorServices.ts

```diff
@@ -930,6 +930,8 @@ namespace ts.server {
 
         /*@internal*/
         delayEnsureProjectForOpenFiles() {
+            // If no open files or no external project, do not schedule
+            if (!this.openFiles.size && !this.externalProjects.length && !this.externalProjectToConfiguredProjectMap.size) return;

```

Comment:
I understand why we skip the work if there are no open files, but not why we're looking at external projects.  Can you please explain?

Reply:
You are right.. this must be friday evening brain fog.. external projects can create configured projects not inferred projects so this doesnt make sense. i am updating the test case..